### PR TITLE
chore: need to move pending api calls back to read function

### PR DESCRIPTION
**Pull request by jebrando**
Followed more towards the win32 implementation and move the pending_api_call back into the read function

### CODE REVIEW

File: linux/src/async_socket_linux.c [hunk 1]
```diff
@@ -117,10 +117,6 @@ static int thread_worker_func(void* parameter)
                 if (recv_context != NULL)
                 {
                     recv_context->on_receive_complete(recv_context->on_receive_complete_context, ASYNC_SOCKET_RECEIVE_ABANDONED, 0);
-
-                    (void)interlocked_decrement(&recv_context->async_socket->pending_api_calls);
```
Comment (anporumb) [inline linux/src/async_socket_linux.c:121]: >pending_api_calls [](http://example.com/codeflow?start=77&length=17)

as there are no specs, what's there to review...


signing off

**Review by anporumb — APPROVED** [review]
:shipit:

**Review by mattdurak — APPROVED** [review]
:shipit:

File: linux/src/async_socket_linux.c (hunk 1 above)
Comment (jebrando) [reply linux/src/async_socket_linux.c:121]: Specs are coming soon.

**Pull request merged**